### PR TITLE
feat: add euclid algorithm

### PR DESCRIPTION
**Pull request by GabinDDL**
add euclid algorithms

### CODE REVIEW

File: content/number-theory/__pycache__/euclid.cpython-310.pyc
Comment (Yag000) [inline content/number-theory/__pycache__/euclid.cpython-310.pyc:1]: Remove this.

File: content/number-theory/euclid.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Author: Unknown
```
Comment (Yag000) [inline content/number-theory/euclid.py]: Bad comment format, see : https://github.com/UPCGM-ICPC/UPCGM-Notebook/pull/6/files

File: content/number-theory/euclid.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# Source: https://www.rookieslab.com/posts/extended-euclid-algorithm-to-find-gcd-bezouts-coefficients-python-cpp-code
+# Description: Finds gcd, and finds two integers $x$ and $y$, such that $ax+by=\gcd(a,b)$. If
+# you just need gcd, use the built in \texttt{\_\_gcd} instead.
+# If $a$ and $b$ are coprime, then $x$ is the inverse of $a \pmod{b}$.
+def gcd(a, b):
```
Comment (Yag000) [inline content/number-theory/euclid.py]: There is a function to perform this on the python standard library, please use that.

**Review by Yag000 — CHANGES_REQUESTED** [review]

**Comment by GabinDDL** [comment]
@Yag000 it's resolved

**Comment by Yag000** [comment]
> @Yag000 it's resolved

Nope, a comment has not been addressed: https://github.com/UPCGM-ICPC/UPCGM-Notebook/pull/13#discussion_r1467413327

And please avoid rebasing when solving issues with your code that have been pointed out in a pr, it makes it harder for the reviewer to understand what you have changed.

**Comment by GabinDDL** [comment]
> > @Yag000 it's resolved
> 
> Nope, a comment has not been addressed: [#13 (comment)](https://github.com/UPCGM-ICPC/UPCGM-Notebook/pull/13#discussion_r1467413327)
> 
> And please avoid rebasing when solving issues with your code that have been pointed out in a pr, it makes it harder for the reviewer to understand what you have changed.

Yes sorry, now it's solved

**Review by Yag000 — APPROVED** [review]

**Comment by Yag000** [comment]
Don't forget to remove the deprecated tests (tests associated with cpp functions removed)

File: content/number-theory/CRT.h
Comment (Yag000) [inline content/number-theory/CRT.h:1]: Why are you removing this?? You have to remove the tests, not the content...

**Review by Yag000 — CHANGES_REQUESTED** [review]

File: content/number-theory/CRT.h
Comment (GabinDDL) [reply content/number-theory/CRT.h:1]: it's true, I put everything back and removed the stress tests

Comment (Yag000) [reply content/number-theory/CRT.h:1]: , since ci is not pasingNot all have been removed

**Review by Yag000 — APPROVED** [review]

**Review by YanisLcn — APPROVED** [review]

**Pull request merged**